distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestCatalogSearchUi.java

```diff
@@ -40,244 +57,432 @@
+                .put(Core.TITLE, Core.TITLE)
+                .put(QUERY_CQL, QUERY)
                 .put("src", ImmutableList.of("source a", "source b"))
                 .build();
 
         List<Map<String, Object>> queries = ImmutableList.of(query);
         Map<String, Object> workspace = ImmutableMap.of(WORKSPACE_QUERIES, queries);
 
-        Response res = expect(asAdmin().body(stringify(workspace)), 201).post(api());
+        Response res = expect(asAdmin().body(stringify(workspace)), 201).post(workspaceApi());
 
-        Map body = parse(res);
-        String id = (String) body.get("id");
-        assertNotNull(id);
+        Map body = parseMap(res);
+        String id = (String) body.get(Core.ID);
+        assertThat(id, notNullValue());
         assertThat(body.get(WORKSPACE_QUERIES), is(queries));
+    }
+
+    @Test
+    public void testLocalAssociation() {
+        setEmptyCswQueryResponse();
+
+        String metacardId1 = ingest(getFileContent("metacard1.xml"), MediaType.APPLICATION_XML);
+        String metacardId2 = ingest(getFileContent("metacard2.xml"), MediaType.APPLICATION_XML);
+        String associationJson = String.format(PUT_ASSOCIATION_STRING,
+                metacardId1,
+                metacardId2,
+                RELATED,
+                AssociationsAttributes.RELATED);
+
+        Response putResponse = expect(asAdmin().body(associationJson), 200).put(
+                associationsApi() + "/" + metacardId1);
+
+        List<Object> associationsPutResponses = parseArray(AssociationsPutResponse.class,
+                putResponse);
+        AssociationsPutResponse associationsPutResponse =
+                (AssociationsPutResponse) associationsPutResponses.get(0);
+        String parentId = ((CharSequenceValue) associationsPutResponse.getParent()
+                .get(Core.ID)).stringValue();
+        assertThat(parentId, is(metacardId1));
+        String childId = ((CharSequenceValue) associationsPutResponse.getChild()
+                .get(Core.ID)).stringValue();
+        assertThat(childId, is(metacardId2));
+        assertThat(associationsPutResponse.getRelationship(), is(RELATED));
+        assertThat(associationsPutResponse.getRelation(), is(AssociationsAttributes.RELATED));
+
+        Response getResponse = expect(asAdmin(), 200).get(associationsApi() + "/" + metacardId1);
+        List<Object> associationGetResponses = parseArray(AssociationsPutResponse.class,
+                getResponse);
+        AssociationsGetResponse associationGetResponse =
+                (AssociationsGetResponse) associationGetResponses.get(0);
+        parentId = ((CharSequenceValue) associationGetResponse.getParent()
+                .get(Core.ID)).stringValue();
+        assertThat(parentId, is(metacardId1));
+        childId = ((CharSequenceValue) associationGetResponse.getChild()
+                .get(Core.ID)).stringValue();
+        assertThat(childId, is(metacardId2));
+
+        clearCatalog();
```

Comment:
just a nit: we could move `clearCatalog` to an `@After` method if we call it after every test.